pgjdbc/src/main/java/org/postgresql/xa/PGXAConnection.java

```diff
@@ -433,27 +463,34 @@ public void commit(Xid xid, boolean onePhase) throws XAException {
   private void commitOnePhase(Xid xid) throws XAException {
     try {
       // Check preconditions
-      if (currentXid == null || !currentXid.equals(xid)) {
-        // In fact, we don't know if xid is bogus, or if it just wasn't associated with this
-        // connection.
+      if (xid.equals(preparedXid)) {
+        throw new PGXAException(GT.tr("One-phase commit called for xid {0} but connection was prepared with xid {1}",
```

Comment:
@ochaloup , Did you really mean `xid.equals(preparedXid)` above?
I think the negation is missing. Can you please clarify the logic here?